apis/v1alpha1/tlsroute_types.go

```diff
@@ -0,0 +1,131 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TLSRouteSpec defines the desired state of TLSRoute
+type TLSRouteSpec struct {
```

Comment:
You can match on the SNI server name at the Listener and use a TCPRoute, so how can we explain when users should do that and when they should use TLSRoute? Maybe we need to address that in the concepts doc rather than in this PR though.

Reply:
That's a great point. I actually didn't think about that case because I assumed that most users would have a Any match for TLS.
I think SNI match here and Host header match in HTTPRoute can have the same semantics here.
What change should we do to this PR?

Reply:
Perhaps we could add some editorial in the doc comment here. Something like "TLSRoute is similar to TCPRoute but can be configured to match against TLS-specific metadata. This allows more flexibility in matching the streams that a Listener is configured to accept."

???